gusto/diagnostics.py

```diff
@@ -53,7 +55,10 @@ def l2(f):
 
     @staticmethod
     def total(f):
-        return assemble(f * dx)
+        if len(f.ufl_shape) == 0:
+            return assemble(f * dx)
+        else:
+            pass
```

Comment:
Hmm, so you could explicitly require that the user states which diagnostics they would like for each field name.

Or you could (sketch, don't take this as gospel), say something like:


```python
class DiagnosticBase(object, metaclass=...):

    @staticmethod
    @abstractmethod
    def computable(field):
        "Is this diagnostic computable on the provided field?"
        pass

class Total(DiagnosticBase):
    def computable(field):
        return len(field.ufl_shape) == 0

...

for diag in available_diagnostics:
    for field in fields:
        if not diag.computable(field)
            continue
       diagnostic_data.add(diag(field))
```